GPU/Vulkan/TextureCacheVulkan.cpp

```diff
@@ -766,11 +679,15 @@ void TextureCacheVulkan::BuildTexture(TexCacheEntry *const entry) {
 	}
 
 	int scaleFactor = standardScaleFactor_;
-	if (scaleFactor > maxScaleFactor_)
-		scaleFactor = maxScaleFactor_;
+	bool hardwareScaling = g_Config.bTexHardwareScaling && uploadCS_ != VK_NULL_HANDLE;
+	if (hardwareScaling) {
+		_assert_(shaderScaleFactor_ != 0);
+		scaleFactor = shaderScaleFactor_;
+	}
 
 	// Rachet down scale factor in low-memory mode.
-	if (lowMemoryMode_) {
+	// TODO: I think really we should just turn it off?
```

Comment:
Maybe.  Nowadays it's most common on mobile:
https://report.ppsspp.org/logs/kind/724

-[Unknown]

Reply:
Leaving it as is in this PR, but yeah.